src/main/java/core/basesyntax/bd/dao/FruitDaoImpl.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.bd.dao;
+
+import core.basesyntax.bd.FruitStorage;
+import core.basesyntax.model.Fruit;
+import java.util.Map;
+
+public class FruitDaoImpl implements FruitDao {
+
+    @Override
+    public void add(Fruit fruit, int amount) {
+        FruitStorage.fruitStorage.put(fruit, amount);
+    }
+
+    @Override
+    public Fruit getKey(Fruit fruit) {
+        return FruitStorage.fruitStorage.keySet().stream()
+                .filter(el -> el.getName().contains(fruit.getName()))
+                .findFirst().orElse(null);
+    }

```

Comment:
seems like you missed some of the comments
please see some comments that may be hidden above